transports/small-webrtc-transport/src/smallWebRTCTransport.ts

```diff
@@ -0,0 +1,630 @@
+            void this.attemptReconnection(true);
+        } else if (this.pc.iceConnectionState === "disconnected") {
+            // Waiting before trying to reconnect to see if it handles it automatically
+            setTimeout(() => {
+                if (this.pc?.iceConnectionState === "disconnected") {
+                    logger.debug("Still disconnected, attempting reconnection.");
+                    void this.attemptReconnection(true);
+                }
+            }, 5000);
+        }
+    }
+
+    private handleReconnectionCompleted() {
+        this.reconnectionAttempts = 0;
+        this.isReconnecting = false;
+    }
+
+    private async attemptReconnection(recreatePeerConnection: boolean = false): Promise<void> {
+        if (this.isReconnecting) {
+            logger.debug("Reconnection already in progress, skipping.");
+            return;
+        }
+        if (this.reconnectionAttempts >= this.maxReconnectionAttempts) {
+            logger.debug("Max reconnection attempts reached. Stopping transport.");
+            this.stop();
+            return;
+        }
+        this.isReconnecting = true;
+        this.reconnectionAttempts++;
+        logger.debug(`Reconnection attempt ${this.reconnectionAttempts}...`);
+        // aiortc it is not working fine when just trying to restart the ice
+        // so in this case we are creating a new peer connection on both sides
+        if (recreatePeerConnection) {
+            const oldPC = this.pc
+            await this.startNewPeerConnection(recreatePeerConnection)
+            if (oldPC) {
+                logger.debug("closing old peer connection")
+                this.closePeerConnection(oldPC)
+            }
+        } else {
+            await this.negotiate();
+        }
+    }
+
+    private async negotiate(recreatePeerConnection: boolean = false): Promise<void> {
+        if (!this.pc) {
+            return Promise.reject('Peer connection is not initialized');
+        }
+
+        try {
+            // Create offer
+            const offer = await this.pc.createOffer();
+            await this.pc.setLocalDescription(offer);
+
+            // Wait for ICE gathering to complete
+            /*await new Promise<void>((resolve) => {
+                if (this.pc!.iceGatheringState === 'complete') {
+                    resolve();
+                } else {
+                    const checkState = () => {
+                        if (this.pc!.iceGatheringState === 'complete') {
+                            this.pc!.removeEventListener('icegatheringstatechange', checkState);
+                            resolve();
+                        }
+                    };
+                    this.pc!.addEventListener('icegatheringstatechange', checkState);
+                }
+            });*/
+
+            let offerSdp = this.pc!.localDescription!;
+            // Filter audio codec
+            if (this.audioCodec && this.audioCodec !== 'default') {
```

Comment:
should we update the type for `audio/videoCodec` to show `'default'` as an option?

Reply:
done.